server: TestAdminAPIQueryPlan failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/eb54cb65ec8da407c8ce5e971157bb1c03efd9e8

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestAdminAPIQueryPlan PKG=github.com/cockroachdb/cockroach/pkg/server TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=856667&tab=buildLog

```
=== RUN   TestAdminAPIQueryPlan
W180823 07:39:01.980953 14741 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180823 07:39:02.168347 14741 server/server.go:830  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180823 07:39:02.168819 14741 base/addr_validation.go:260  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180823 07:39:02.168920 14741 base/addr_validation.go:300  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180823 07:39:02.202026 14741 server/config.go:496  [n?] 1 storage engine initialized
I180823 07:39:02.202127 14741 server/config.go:499  [n?] RocksDB cache size: 128 MiB
I180823 07:39:02.202169 14741 server/config.go:499  [n?] store 0: in-memory, size 0 B
I180823 07:39:02.358764 14741 server/node.go:373  [n?] **** cluster 4563c89a-b8bf-4cda-b160-815eb499fa7b has been created
I180823 07:39:02.358918 14741 server/server.go:1401  [n?] **** add additional nodes by specifying --join=127.0.0.1:38833
I180823 07:39:02.359924 14741 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38833" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:12 > build_tag:"v2.1.0-alpha.20180702-1924-geb54cb6" started_at:1535009942359606618 
I180823 07:39:02.410349 14741 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180823 07:39:02.416528 14741 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180823 07:39:02.417663 14741 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I180823 07:39:02.418214 14741 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "4563c89a-b8bf-4cda-b160-815eb499fa7b"
I180823 07:39:02.420141 14741 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180823 07:39:02.423933 14741 server/status/recorder.go:652  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180823 07:39:02.424408 14741 server/server.go:1807  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180823 07:39:02.438703 14970 server/status/recorder.go:652  [n1,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180823 07:39:02.441962 14741 server/server.go:1538  [n1] starting https server at 127.0.0.1:39547 (use: 127.0.0.1:39547)
I180823 07:39:02.446738 14982 storage/replica_command.go:289  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180823 07:39:02.469657 14741 server/server.go:1540  [n1] starting grpc/postgres server at 127.0.0.1:38833
I180823 07:39:02.469773 14741 server/server.go:1541  [n1] advertising CockroachDB node at 127.0.0.1:38833
W180823 07:39:02.614499 14762 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=f50f05df key=/Table/SystemConfigSpan/Start rw=true pri=0.00249380 iso=SERIALIZABLE stat=PENDING epo=0 ts=1535009942.502820692,0 orig=1535009942.502820692,0 max=1535009942.502820692,0 wto=false rop=false seq=6
I180823 07:39:02.615200 14479 storage/replica_command.go:289  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180823 07:39:02.666024 14760 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180823 07:39:02.715673 15010 storage/replica_command.go:289  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180823 07:39:03.070798 14952 storage/replica_command.go:289  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180823 07:39:03.195443 15042 storage/replica_command.go:289  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180823 07:39:03.289625 14463 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:root}
I180823 07:39:03.330713 15047 storage/replica_command.go:289  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180823 07:39:03.449077 15058 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180823 07:39:03.492661 15033 storage/replica_command.go:289  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180823 07:39:03.596919 15065 storage/replica_command.go:289  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180823 07:39:03.651445 15050 rpc/nodedialer/nodedialer.go:92  [consistencyChecker,n1,s1,r1/1:/{Min-System/}] connection to n1 established
I180823 07:39:03.721080 15080 storage/replica_command.go:289  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180823 07:39:03.813216 15110 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:root}
I180823 07:39:03.861180 15128 storage/replica_command.go:289  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180823 07:39:03.970081 15067 storage/replica_command.go:289  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180823 07:39:03.993576 15125 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180823 07:39:04.036906 15071 storage/replica_command.go:289  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180823 07:39:04.087366 15114 storage/replica_command.go:289  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180823 07:39:04.105370 15083 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180823 07:39:04.162955 15218 storage/replica_command.go:289  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180823 07:39:04.186095 14741 server/server.go:1594  [n1] done ensuring all necessary migrations have run
I180823 07:39:04.186213 14741 server/server.go:1597  [n1] serving sql connections
I180823 07:39:04.255146 15220 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180823 07:39:04.259898 15222 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:38833} Attrs: Locality: ServerVersion:2.0-12 BuildTag:v2.1.0-alpha.20180702-1924-geb54cb6 StartedAt:1535009942359606618 LocalityAddress:[]} ClusterID:4563c89a-b8bf-4cda-b160-815eb499fa7b StartedAt:1535009942359606618 LastUp:1535009942359606618}
I180823 07:39:04.286545 15159 storage/replica_command.go:289  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180823 07:39:04.387013 15166 storage/replica_command.go:289  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180823 07:39:04.484268 15104 storage/replica_command.go:289  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180823 07:39:04.572186 15199 storage/replica_command.go:289  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180823 07:39:04.643657 15268 storage/replica_command.go:289  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180823 07:39:04.707696 15275 storage/replica_command.go:289  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180823 07:39:04.766374 15302 storage/replica_command.go:289  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180823 07:39:05.025875 15255 storage/replica_command.go:289  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180823 07:39:05.034487 15350 sql/event_log.go:126  [n1,client=127.0.0.1:39344,user=root] Event: "create_database", target: 52, info: {DatabaseName:api_test Statement:CREATE DATABASE api_test User:root}
I180823 07:39:05.122921 15350 sql/event_log.go:126  [n1,client=127.0.0.1:39344,user=root] Event: "create_table", target: 53, info: {TableName:api_test.public.t1 Statement:CREATE TABLE api_test.public.t1 (id INT PRIMARY KEY, name STRING) User:root}
I180823 07:39:05.149652 15318 storage/replica_command.go:289  [split,n1,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180823 07:39:05.215571 15350 sql/event_log.go:126  [n1,client=127.0.0.1:39344,user=root] Event: "create_table", target: 54, info: {TableName:api_test.public.t2 Statement:CREATE TABLE api_test.public.t2 (id INT PRIMARY KEY, name STRING) User:root}
I180823 07:39:05.303779 15263 storage/replica_command.go:289  [split,n1,s1,r24/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r25]
==================
WARNING: DATA RACE
Read at 0x00c421835cd8 by goroutine 310:
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPlanForNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:162 +0x21b
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).wrapPlan.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2470 +0x193
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:117 +0x7e2f
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:411 +0x2e3f
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitConcrete()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:107 +0xab
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:477 +0x1603
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).wrapPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2445 +0x3d5
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPlanForNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2399 +0x143
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:729 +0x16c
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:966 +0x57f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:808 +0x8f0
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:401 +0x108d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1135 +0x1ed4
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:210 +0x97

Previous write at 0x00c421835cd8 by goroutine 81:
  github.com/cockroachdb/cockroach/pkg/sql.(*explainDistSQLNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/explain_distsql.go:75 +0x348
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:132 +0x7d1b
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:126 +0x154
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:263 +0x7c
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartSync()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:594 +0x329
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:245 +0xb58
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:735 +0x297
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:966 +0x57f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:808 +0x8f0
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:401 +0x108d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:210 +0x97

Goroutine 310 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:199 +0x49a
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:449 +0x54d
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:292 +0x1d1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).insertRangeLogEvent()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:99 +0x38f
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).logSplit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/log.go:121 +0x353
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:329 +0x422
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:585 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:692 +0xf6
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:584 +0x13b
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).adminSplitWithDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:292 +0xcc7
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).processAttempt()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:149 +0x1c8
  github.com/cockroachdb/cockroach/pkg/storage.(*splitQueue).process()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/split_queue.go:127 +0x93
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processReplica()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:744 +0x313
  github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1.2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:627 +0xdd
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf3

Goroutine 81 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:199 +0x49a
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:449 +0x54d
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:244 +0x111
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryWithSessionArgs()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:261 +0x127
  github.com/cockroachdb/cockroach/pkg/server.(*adminServer).QueryPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/admin.go:1254 +0x31b
  github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_QueryPlan_Handler.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1862 +0xa4
  github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:48 +0xc23
  github.com/cockroachdb/cockroach/pkg/rpc.NewServerWithInterceptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:166 +0x10c
  github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_QueryPlan_Handler()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1864 +0x1e3
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).processUnaryRPC()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1011 +0xde7
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleStream()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1249 +0x1364
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:680 +0xac
==================
I180823 07:39:05.488325 14741 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
I180823 07:39:05.489517 14987 kv/transport_race.go:91  transport race promotion: ran 72 iterations on up to 932 requests
--- FAIL: TestAdminAPIQueryPlan (3.61s)
	testing.go:730: race detected during execution of test

```

--- a/pkg/sql/explain_distsql.go
+++ b/pkg/sql/explain_distsql.go
@@ -69,12 +69,6 @@ func (n *explainDistSQLNode) startExec(params runParams) error {
 	planCtx.stmtType = n.stmtType
 	planCtx.validExtendedEvalCtx = true
 
-	// This stanza ensures that EXPLAIN(DISTSQL) won't include metadata test
-	// senders or receivers.
-	curTol := distSQLPlanner.metadataTestTolerance
-	distSQLPlanner.metadataTestTolerance = distsqlrun.On
-	defer func() { distSQLPlanner.metadataTestTolerance = curTol }()
-
 	plan, err := distSQLPlanner.createPlanForNode(&planCtx, n.plan)
 	if err != nil {
 		return err

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -958,7 +958,7 @@ func (t *logicTest) setup(cfg testClusterConfig) {
 		distSQLKnobs.MemoryLimitBytes = 1
 	}
 	if cfg.distSQLMetadataTestEnabled {
-		distSQLKnobs.MetadataTestLevel = distsqlrun.NoExplain
+		distSQLKnobs.MetadataTestLevel = distsqlrun.On
 	}
 	params.ServerArgs.Knobs.DistSQL = distSQLKnobs
 

